frontend/src/searchform/IngredientList.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from "react";
+import "./IngredientList.css";
+
+import { IIngredient } from "../../../types";
+
+import * as API from "../api";
+
+const loadAllIngredients = API.getIngredients();
+
+async function getRecommendation(search: string, exclude: IIngredient[], cancellationToken: { cancelled: boolean }): Promise<IIngredient | null> {
+    const ingredients = (await loadAllIngredients).ingredients;
+
+    if(cancellationToken.cancelled)
+        throw new Error("cancelled");
+
+    let best = null;
+    
+    for(const ingredient of ingredients) {
+        if(exclude.includes(ingredient)) continue;
+        if(!ingredient.name.startsWith(search)) continue;
```

Comment:
Search should be case-insensitive to simplify searching for ingredients with proper names ("California brandy", "Dr Pepper") and to better fit touch keyboards that capitalize by default.

We might also want to remove accents and other special characters. Searching for "Curaçao" or "Bénédectine" is cumbersome otherwise. [This solution](https://stackoverflow.com/questions/990904/remove-accents-diacritics-in-a-string-in-javascript) is a little hacky, but sufficiently standardized and should work just fine.